### ISSUE TITLE
Feature enhancement: Include to POI search other Map features as well

### ISSUE THREAD

**Issue by HTenkanen** [gboeing/osmnx#190]
Hi @gboeing, one of my students just pointed out that the poi search is missing many useful poi types such as:

- [Shop](https://wiki.openstreetmap.org/wiki/Map_Features#Shop)
- [Craft](https://wiki.openstreetmap.org/wiki/Map_Features#Craft)
- [Emergency](https://wiki.openstreetmap.org/wiki/Map_Features#Emergency)
- [Historic](https://wiki.openstreetmap.org/wiki/Map_Features#Historic)
- [Leisure](https://wiki.openstreetmap.org/wiki/Map_Features#Leisure)

I can work on this one to get these also included but there are a few different ways of building the logic of including these different types for the search. 

The functions could work in a way that:

1. There is e.g. a parameter for the key words that the user wants to search and then the search would be finding the POIs from all these different sources. This would be the most simple one for the user but it lacks the possibility to exclude some sources for POIs.

2. Second option is that we provide parameters for all of these different sources and user defines lists for all the sources s/he wants to search. E.g. `amenities = ['grocery'], shops=['grocery']` etc. This might be a bit unpractical though.

3. The last one (which is the logic I vote for) would be to use flags for different sources. E.g. the search criteria would be passed with parameter `keys = ['grocery']` and then the user could define with boolean parameters the sources s/he wants to search the POIs from, e.g. `amenities = True, shops=True, emergency=False, leisure=True` etc. Other possibility would be to pass a list of values like `sources = ['amenities', 'shops', 'leisure']`. However, the flags might be more stable way of doing this. 

Do you have any thoughts about this?

Cheers,
Henrikki

**Comment by gboeing** [comment]
I like the idea of the 3rd option as it seems most powerful. What about passing a dict instead like:

```python
{'amenities' : ['grocery', 'restaurant', 'school'],
 'leisure' : 'park'
 'shops' : True,
 'emergency' : False}
```

This dict would signify that we ignore all `emergency` elements, retrieve all `shops` elements, retrieve `leisure` elements where leisure value = `park`, and also retrieve all `amenities` where amenity value = `grocery` or `restaurant` or `school`.

So, in the argument dict, `key`:`False` means ignore all with this key, `key`:`True` means retrieve all with this key, `key`:`string` means retrieve all where this key has value string, and `key : [string1, string2]` means retrieve all where this key has value of any string in the list.

Thoughts?

**Comment by HTenkanen** [comment]
Yes, that sounds like a good idea. 👍 It's flexible and easy to understand. 

What should we use as default values? Should we just fetch everything? It might produce quite a bit of results e.g. in Manhattan 😛 In the current version it will fetch everything but only from amenities.

**Comment by gboeing** [comment]
Maybe the default value could be `{'amenities' : True}` to maintain the current default functionality then?

**Comment by HTenkanen** [comment]
@gboeing Yes that is perhaps the best option here. I could also work on short tutorial notebook for fetching pois as it is still missing, and would perhaps be good to have it available as well.

**Comment by gboeing** [comment]
@HTenkanen that would be fantastic.

**Comment by gboeing** [comment]
@HTenkanen just checking back if this is still active.

**Comment by YATPKU** [comment]
Hello Sirs,
@HTenkanen , @gboeing 
I am a newbie, and I was trying to save amenity as shapefile for  GIS but failed. May you give me some advice?
Thank you very much!

**Comment by HTenkanen** [comment]
Hi @YATPKU ! 

If you are new to using the package, you could take a look at these tutorials that shows how to:

1. [Download OSM data with OSMnx](https://automating-gis-processes.github.io/2018/notebooks/L6/retrieve_osm_data.html)
2. [How to save OSM data using OSMnx into Shapefiles](https://automating-gis-processes.github.io/2018/notebooks/L6/network-analysis.html#Saving-shortest-paths-to-disk)

There is no exact example showing how to save POIs into Shapefile but following the example from the link number 2 (at the bottom of that page), you should get an idea how to save data from OSM into Shapefile. 

Cheers,
Henrikki

**Comment by YATPKU** [comment]
@HTenkanen 

Thanks for your prompt reply. I really appreciate.
Thanks!

**Comment by gboeing** [comment]
@YATPKU OSMnx [returns POIs](https://osmnx.readthedocs.io/en/stable/osmnx.html#osmnx.pois.pois_from_place) as a geopandas [GeoDataFrame](http://geopandas.org/reference/geopandas.GeoDataFrame.html), which can be directly [saved](http://geopandas.org/io.html#writing-spatial-data) to disk as a shapefile.

**Comment by YATPKU** [comment]
@gboeing 
Thank Prof. for your reply. I will try it and come back to you if it does not work.
Regards,

**Comment by gboeing** [comment]
Moving this conversation to PR #240 (see also recent discussion in #116).

**Comment by gboeing** [comment]
resolved in #449
